api/test/api/adapter/test_cognito_adapter.py

```diff
@@ -247,6 +248,9 @@ def test_throws_error_when_client_app_name_has_not_been_changed_from_placeholder
 
 
 class TestCognitoAdapterUsers(BaseCognitoAdapter):
+    @mock.patch.dict(
+        os.environ, {"CUSTOM_USERNAME_REGEX": "[a-zA-Z][a-zA-Z0-9@._-]{2,127}"}
```

Comment:
I don't think we want to patch the variable here but instead enforce it by ensuring it's set correctly within the environment variables when running the tests.

We will want a test where we patch it to be None and ensure we test the counter-factual of when they don't set it and thus we just fallback to the default validation